### PR TITLE
SCMOD-5877: Removed generic exception from interface

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**
https://portal.digitalsafe.net/browse/SCMOD-5877

### CODE REVIEW

File: caf-audit/src/main/java/com/hpe/caf/auditing/exception/AuditFailureException.java [hunk 1]
```diff
@@ -13,14 +13,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hpe.caf.auditing.webserviceclient;
+package com.hpe.caf.auditing.exception;
 
-public class WebServiceClientException extends Exception {
-    public WebServiceClientException(String message) {
+public class AuditFailureException extends Exception
```
Comment (dermot-hardy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/exception/AuditFailureException.java]: How about `AuditingException`?  Having `Failure` in there seems redundant.

File: caf-audit-binding-webservice/src/main/java/com/hpe/caf/auditing/webserviceclient/WebServiceClientAuditEventBuilder.java [hunk 2]
```diff
@@ -193,7 +194,7 @@ public void send() throws IOException, WebServiceClientException {
         if (auditEventJson == null || auditEventJson.isEmpty()) {
             String errorMessage = "No Audit Event JSON to send to the Audit WebService";
             LOG.error(errorMessage);
-            throw new WebServiceClientException(errorMessage);
+            throw new AuditFailureException(errorMessage);
```
Comment (dermot-hardy) [inline caf-audit-binding-webservice/src/main/java/com/hpe/caf/auditing/webserviceclient/WebServiceClientAuditEventBuilder.java]: Is this not just library misuse?  If so then it should be some type of unchecked exception thrown here.

File: caf-audit/src/main/java/com/hpe/caf/auditing/noop/NoopAuditEventBuilder.java [hunk 3]
```diff
@@ -97,7 +99,7 @@ public void addEventParameter(final String name, final String columnName, final
     }
 
     @Override
-    public void send() throws Exception
+    public void send() throws IOException, AuditFailureException
```
Comment (dermot-hardy) [inline caf-audit/src/main/java/com/hpe/caf/auditing/noop/NoopAuditEventBuilder.java]: You can actually drop the exception specification altogether here.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
We will also need plugin changes but they can be a separate PR.

**Comment by Anthony-McGreevy** [comment]
@dermot-hardy Can you look at this again. This now includes your requested changes and the changes required for the plugin.

**Comment by Anthony-McGreevy** [comment]
@dermot-hardy  Can you try and look at this today if you have time.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**